x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/openai/embeddings/OpenAiEmbeddingsModel.java

```diff
@@ -21,6 +21,15 @@
 
 public class OpenAiEmbeddingsModel extends OpenAiModel {
 
+    public static OpenAiEmbeddingsModel of(OpenAiEmbeddingsModel model, Map<String, Object> taskSettings) {
```

Comment:
Thanks for this change! I think this implementation is much clearer with regards to the object references